### PR TITLE
ENH: Adds GenericMotor.load_from_eng_file() method

### PR DESCRIPTION
**Pull request by juliomachad0**
## Pull request type
- [x] Code changes (bugfix, features)

## Checklist
- [x] Lint (`black rocketpy/ tests/`) has passed locally 
- [x] All tests (`pytest tests -m slow --runslow`) have passed locally

## Current behavior
the static method to create motors from eng files did not exist.

## New behavior
method created and working

## Breaking change
- [x] No

### CODE REVIEW

File: rocketpy/motors/motor.py [hunk 1]
```diff
@@ -1179,6 +1214,145 @@ def __init__(
         self.prints = _MotorPrints(self)
         self.plots = _MotorPlots(self)
 
+    @staticmethod
+    def load_from_eng_file(
+        file_name,
+        nozzle_radius,
```
Comment (Gui-FernandesBR) [inline rocketpy/motors/motor.py:1220]: My suggestion:

Don't make `nozzle_radius` a mandatory argument. If user doesn`t provide it, set the noozle radius as a percentage of the chamber radius. Let this clear on the documentation.


```suggestion
        nozzle_radius = None,
```

File: rocketpy/motors/motor.py [hunk 2]
```diff
@@ -1179,6 +1214,145 @@ def __init__(
...
+
+        thrust = Function(thrust_source, "Time (s)", "Thrust (N)", "linear", "zero")
+
+        # handle eng parameters
+
+        if chamber_radius:
+            if not isinstance(chamber_radius, (int, float)):
+                raise ValueError("Chamber radius must be a number.")
+        else:
+            chamber_radius = float(description[1])  # get motor diameter
```
Comment (Gui-FernandesBR) [inline rocketpy/motors/motor.py:1314]: Avoid creating too much oinput validations, it's against python's essence. 

```suggestion
        if not chamber_radius:
            chamber_radius = float(description[1])  # get motor diameter
```

File: rocketpy/motors/motor.py [hunk 3]
```diff
@@ -1179,6 +1214,145 @@ def __init__(
...
+
+        if chamber_height:
+            if not isinstance(chamber_height, (int, float)):
+                raise ValueError("Chamber height must be a number.")
+        else:
+            chamber_height = float(description[2])  # get motor length
```
Comment (Gui-FernandesBR) [inline rocketpy/motors/motor.py:1320]:
```suggestion
        if not chamber_height:
            chamber_height = float(description[2])  # get motor length
```

File: rocketpy/motors/motor.py [hunk 4]
```diff
@@ -1179,6 +1214,145 @@ def __init__(
...
+            nozzle_radius=nozzle_radius,
+            dry_mass=dry_mass,
+            center_of_dry_mass_position=center_of_dry_mass_position,
+            dry_inertia=dry_inertia,
+            nozzle_position=nozzle_position,
+            reshape_thrust_curve=reshape_thrust_curve,
+            interpolation_method=interpolation_method,
+            coordinate_system_orientation=coordinate_system_orientation,
+        )
+        return gen_motor
```
Comment (Gui-FernandesBR) [inline rocketpy/motors/motor.py:1354]: Save lines, save time.

```suggestion
        return GenericMotor(
            thrust_source=thrust,
            burn_time=burn_time,
            chamber_radius=chamber_radius,
            chamber_height=chamber_height,
            chamber_position=chamber_position,
            propellant_initial_mass=propellant_initial_mass,
            nozzle_radius=nozzle_radius,
            dry_mass=dry_mass,
            center_of_dry_mass_position=center_of_dry_mass_position,
            dry_inertia=dry_inertia,
            nozzle_position=nozzle_position,
            reshape_thrust_curve=reshape_thrust_curve,
            interpolation_method=interpolation_method,
            coordinate_system_orientation=coordinate_system_orientation,
        )
```

File: rocketpy/motors/motor.py [hunk 5]
```diff
@@ -1179,6 +1214,145 @@ def __init__(
...
+
+        if dry_mass:
+            if not isinstance(dry_mass, (int, float)):
+                raise ValueError("Dry mass must be a number.")
+        else:
+            total_mass = float(description[-2])
+            dry_mass = total_mass - propellant_initial_mass
+
+        e_vel = thrust.integral(*(0.0, 4.897)) / propellant_initial_mass
+        print(f"exhaust velocity: {e_vel:.2f} m/s ")
```
Comment (Gui-FernandesBR) [inline rocketpy/motors/motor.py:1336]: I believe you should delete this.


```suggestion
```

File: rocketpy/motors/motor.py [hunk 6]
```diff
@@ -1179,6 +1214,145 @@ def __init__(
...
+                raise ValueError("Propellant initial mass must be a number.")
+        else:
+            propellant_initial_mass = float(description[-3])
+
+        if dry_mass:
+            if not isinstance(dry_mass, (int, float)):
+                raise ValueError("Dry mass must be a number.")
+        else:
+            total_mass = float(description[-2])
+            dry_mass = total_mass - propellant_initial_mass
```
Comment (Gui-FernandesBR) [inline rocketpy/motors/motor.py:1333]:
```suggestion
        if not dry_mass:
            total_mass = float(description[-2])
            dry_mass = total_mass - propellant_initial_mass
```

File: rocketpy/motors/motor.py [hunk 7]
```diff
@@ -1179,6 +1214,145 @@ def __init__(
...
+            if not isinstance(chamber_height, (int, float)):
+                raise ValueError("Chamber height must be a number.")
+        else:
+            chamber_height = float(description[2])  # get motor length
+
+        if propellant_initial_mass:
+            if not isinstance(propellant_initial_mass, (int, float)):
+                raise ValueError("Propellant initial mass must be a number.")
+        else:
+            propellant_initial_mass = float(description[-3])
```
Comment (Gui-FernandesBR) [inline rocketpy/motors/motor.py:1326]:
```suggestion
        if not propellant_initial_mass:
            propellant_initial_mass = float(description[-3])
```

File: rocketpy/motors/motor.py [hunk 8]
```diff
@@ -1179,6 +1214,145 @@ def __init__(
...
+            is defined by the motor's axis of symmetry. The origin of the
+            coordinate system may be placed anywhere along such axis, such as
+            at the nozzle area, and must be kept the same for all other
+            positions specified. Options are "nozzle_to_combustion_chamber" and
+            "combustion_chamber_to_nozzle". Default is
+            "nozzle_to_combustion_chamber".
+        Returns
+        -------
+        Generic Motor object
+        """
```
Comment (Gui-FernandesBR) [inline rocketpy/motors/motor.py:1297]: give an empty space.

Are you using vscode extensions for autogenerating docstrings? That should help.

```suggestion
            "nozzle_to_combustion_chamber".

        Returns
        -------
        Generic Motor object
        """
```

File: rocketpy/motors/motor.py [hunk 9]
```diff
@@ -1179,6 +1214,145 @@ def __init__(
...
+        if isinstance(file_name, str):
```
Comment (Gui-FernandesBR) [inline rocketpy/motors/motor.py:1298]: Only strings? 

Are you aware of Path module form python?
https://docs.python.org/3/library/os.path.html

You can create somekind of validation like: "if filename ends with .eng"

**Review by Gui-FernandesBR — CHANGES_REQUESTED** [review]

File: rocketpy/motors/motor.py (hunk 1 above)
Comment (juliomachad0) [reply rocketpy/motors/motor.py:1220]: ok

File: rocketpy/motors/motor.py (hunk 5 above)
Comment (juliomachad0) [reply rocketpy/motors/motor.py:1336]: True. I was doing some tests, I forgot to delete it

File: rocketpy/motors/motor.py (hunk 9 above)
Comment (juliomachad0) [reply rocketpy/motors/motor.py]: done

**Review by Gui-FernandesBR — APPROVED** [review]
Really solid work, @juliomachad0 !! The new documentation page was really necessary and I believe it is going to help new users quite a lot!

**Pull request merged**